superset-frontend/src/components/Chart/DrillBy/DrillByChart.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import {
+  Behavior,
+  BinaryQueryObjectFilterClause,
+  Column,
+  css,
+  SuperChart,
+  useTheme,
+} from '@superset-ui/core';
+import { simpleFilterToAdhoc } from 'src/utils/simpleFilterToAdhoc';
+import { getChartDataRequest } from 'src/components/Chart/chartAction';
+import Loading from 'src/components/Loading';
+
+interface DrillByChartProps {
+  column?: Column;
+  filters?: BinaryQueryObjectFilterClause[];
+  formData: { [key: string]: any; viz_type: string };
+  groupbyFieldName?: string;
+}
+
+export default function DrillByChart({
+  column,
+  filters,
+  formData,
+  groupbyFieldName,
+}: DrillByChartProps) {
+  let updatedFormData = formData;
+  let groupBy: any = [];
+  const theme = useTheme();
+  const [chartDataResult, setChartDataResult] = useState();
+
+  if (groupbyFieldName && column) {
+    groupBy = Array.isArray(formData[groupbyFieldName])
+      ? [column.column_name]
+      : column.column_name;
+  }
+
+  if (filters) {
+    const adhocFilter = simpleFilterToAdhoc({
+      ...filters[0],
+      isExtra: true,
+    });
+    updatedFormData = {
+      ...formData,
+      adhoc_filters: [...formData.adhoc_filters, adhocFilter],
+      [groupbyFieldName || 'groupby']: groupBy,
+    };
+  }
+
+  useEffect(() => {
+    getChartDataRequest({
+      formData: updatedFormData,
+      requestParams: { dashboardId: 0 },
+    }).then(({ json }) => {
+      setChartDataResult(json.result);
+    });
+  }, []);
+
+  return (
+    <div
+      css={css`
+        align-items: center;
+        display: flex;
+        height: 100%;
+        justify-content: center;
+      `}
+    >
+      {chartDataResult ? (
+        <SuperChart
+          behaviors={[Behavior.INTERACTIVE_CHART]}
```

Comment:
`INTERACTIVE_CHART` is for cross filters, and in the drill by modal we only want to support further drill by... but let's leave it for a separate pr to clean up 🙂